File: pilha.c
#include <stdio.h>

#define CAP 1000

struct pilha {
	int a[CAP];
	int topo;
};

int iniciaPilha(struct pilha *p) {
	p->topo = -1;
}

int pilhaVazia(struct pilha p) {
	if (p.topo == -1) return 1;
	else return 0;
}

int topoPilha(struct pilha p) {
	return p.a[p.topo];
}

int empilha(struct pilha *p, int x) {
	if (p->topo != (CAP - 1)) {
		p->topo++;
		p->a[p->topo] = x;
		return 0;
	} else {
		printf("Overlhow!");
		return -1;
	}
}

int desempilha(struct pilha *p) {
	int valor;
	if (p->topo != -1) {
		valor = p->a[p->topo];
		p->topo--;
		return valor;
	} else {
		printf("Underflow!");
		return 0;
	}
}

int main() {
	struct pilha stack;

	iniciaPilha(&stack);

	empilha(&stack, 100);
	empilha(&stack, 20);

	desempilha(&stack);

	printf("O topo é %d\n", topoPilha(stack));

	printf("%d\n", (-2) % CAP);

	return 0;
}

File: alunos/Vinicius/Pilha/Pilha.java
public class Pilha {
    private final int CAPACIDADE = 1000;
	private int [] vetor = new int[CAPACIDADE];
	private int topo;
	
	public Pilha () {
		this.topo = -1;
	}
	
	//Retorna se a pilha está vazia ou não
	public boolean estaVazia() {
		if(topo == -1) return true;
		else return false;
	}
	
	//Retorna o topo da pilha
	public int topoPilha() {
		return this.topo;
	}
	
	//Método de inserção na pilha
	public int empilha(int x){
		if(this.topo != (CAPACIDADE -1)) {
			this.topo++;
			this.vetor[this.topo] = x;
			return 0;
		} else {
			System.out.println("Overflow!");
			return -1;
		}
	}
	
	//Método de remoção da pilha
	public int desempilha() {
		int valor;
		if(this.topo != -1) {
			valor = this.vetor[this.topo];
			this.topo--;
			return valor;
		} else {
			System.out.println("Underflow!");
			return 0;
		}
	}
	//Imprime a pilha
		public void mostraPilha() {
			String elementos = "";
			for(int i = 0; i <= this.topo;i++) {
				elementos += this.vetor[i] +  "\n";
			}
			
			System.out.print(elementos);
		}
}
